Simplify type signature for spawn() function. Addresses #1567 (#1570)
davidkpiano/xstate: .changeset/blue-ladybugs-retire.md

---
'xstate': patch
---

> The return type of `spawn(machine)` will now be `Actor<State<TContext, TEvent>, TEvent>`, which is compatible with `Interpreter<...>`.

Reviewer: ```suggestion
The return type of `spawn(machine)` will now be `Actor<State<TContext, TEvent>, TEvent>`, which is a supertype of `Interpreter<...>`.
```
